crypto/thr_id.c

```diff
@@ -199,6 +204,8 @@ void CRYPTO_THREADID_current(CRYPTO_THREADID *id)
     /* Else pick a backup */
 #if defined(OPENSSL_SYS_WIN32)
     CRYPTO_THREADID_set_numeric(id, (unsigned long)GetCurrentThreadId());
+#elif defined(__unix__) || defined(unix)
+    CRYPTO_THREADID_set_numeric(id, (unsigned long)pthread_self());
```

Comment:
Not sure. We had a threaded test causing a malloc crash in openssl which got fixed when we used `pthread_self()` instead.